src/backend/distributed/commands/parameter.c

```diff
@@ -0,0 +1,178 @@
+#include "postgres.h"
+
+#include "pg_version_constants.h"
+#if PG_VERSION_NUM >= PG_VERSION_15
+#include "access/genam.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_parameter_acl.h"
+#include "commands/defrem.h"
+#include "utils/acl.h"
+#include "utils/builtins.h"
+#include "utils/syscache.h"
+
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/grant_utils.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_sync.h"
+
+static List * GenerateGrantOnParameterFromAclItem(char *parameterName, AclItem *aclItem);
+static bool HasAclGrantOption(AclItem *aclItem, AclMode aclMode);
+static void CheckPermissionsAndGrants(AclItem *aclItem, AclMode modes[], int numModes);
+static void CheckAndAppendQuery(List **queries, AclItem *aclItem, Oid granteeOid,
+								char *parameterName, AclMode mode, char *modeStr);
+
+
+List *
+PostprocessGrantParameterStmt(Node *node, const char *queryString)
+{
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	EnsurePropagationToCoordinator();
+
+	char *command = DeparseTreeNode(node);
+
+	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
+								(void *) command,
+								ENABLE_DDL_PROPAGATION);
+
+	return NontransactionalNodeDDLTaskList(REMOTE_NODES, commands);
```

Comment:
why non-transactional?

Reply:
It is not required to be non-transactional. Changed it with NodeDDLTask